octoprint_simplyprint/websocket/simplyprint.py

```diff
@@ -1012,7 +1012,7 @@ async def _handle_ai_snapshot(self, eventtime: float) -> float:
             headers = {"User-Agent": "Mozilla/5.0"}
             data = json.dumps({"api_key": self.settings.get(["printer_token"]), "image_array": img_data, "interval": ai_interval}).encode('utf8')
             response = requests.get("https://ai.simplyprint.io/api/v2/infer", data=data, headers=headers)
```

Comment:
Hi Jim, I just thought I would drop in and leave a comment.  The call  on line 1014 to `requests.get()` and the call on line 1011 to `extract_image()` will block the main thread, so its probably a good idea to run them in the default thread pool executor via `self._loop.run_in_executor()`, there is an example on line 1000.  [It can't take keyword arguments](https://www.tornadoweb.org/en/stable/ioloop.html#tornado.ioloop.IOLoop.run_in_executor), so you will probably need to wrap the `requests` call with `functools.partial`.

Reply:
I thought that would be in it's own thread using the FlexTimer. I guess that doesn't work the same as octoprint.util.RepeatedTimer?

Reply:
Just to verify I'm doing this right Eric, something like this?

```
img_data = await self._loop.run_in_executor(None, self.webcam_stream.extract_image)
response = await self._loop.run_in_executor(None, functools.partial(requests.get, ["https://ai.simplyprint.io/api/v2/infer"], {"data": data, "headers": headers}))
```

Reply:
> I thought that would be in it's own thread using the FlexTimer. I guess that doesn't work the same as octoprint.util.RepeatedTimer?

Ah, I see.  The FlexTimer does run on the main thread since the timer function is a coroutine.  Without  getting too far into the weeds, I basically took it straight from Moonraker since SimplyPrint's websocket client is close to a straight port from Moonraker's.   This is a different pattern for Octoprint plugins since it relies on the underlying asyncio eventloop for concurrency.  I'm not sure if any other plugin uses the IOLoop/eventloop.